core/ledger/kvledger/upgrade_dbs.go

```diff
@@ -24,29 +33,44 @@ func UpgradeDBs(rootFSPath string) error {
 	defer fileLock.Unlock()
 
 	logger.Infof("Ledger data folder from config = [%s]", rootFSPath)
+	dbPath := LedgerProviderPath(rootFSPath)
+	db := leveldbhelper.CreateDB(&leveldbhelper.Conf{DBPath: dbPath})
+	db.Open()
+	defer db.Close()
 
-	// upgrades idStore (ledgerProvider) and drop databases
-	if err := UpgradeIDStoreFormat(rootFSPath); err != nil {
-		return err
+	if emptydb, _ := db.IsEmpty(); emptydb {
+		return errors.Errorf("ledger database %s is empty. Make sure core.yaml has the correct ledger path.", dbPath)
```

Comment:
it will be empty when the peer has joined no channel. Anyway, as I mentioned earlier, it is a remote corner case.

Reply:
I thought this message may alert the user in case the ledger db should not be empty. If this is not the case, we can remove the checking. Otherwise, it is ok for the message to disappear if the user starts the peer once in v2.0 and re-runs upgrade-dbs. We may change the error to a warning message though.

Reply:
I keep the check because otherwise we cannot tell why formatKey does not exist (either 1.x version or empty db).